internal/subscription/core.go

```diff
@@ -103,9 +101,37 @@ func (c *Core) CreateSubscription(ctx context.Context, m *Model) error {
 		})
 
 		if err != nil {
-			logger.Ctx(ctx).Errorw("failed to create deadletter topic for subscription", "name", m.GetDeadLetterTopic(), "error", err.Error())
+			logger.Ctx(ctx).Errorw("failed to create dead letter topic for subscription", "name", m.GetDeadLetterTopic(), "error", err.Error())
 			return err
 		}
+
+		// this creates the needed retry/delay topic configs
+		m.DelayConfig, err = NewDelayConfig(m)
+		if err != nil {
+			logger.Ctx(ctx).Errorw("failed to init delay config", "error", err.Error())
+			return err
+		}
+
+		// create all the needed delay topics
+		for _, delayTopic := range m.DelayConfig.DelayTopics {
```

Comment:
The part where we create all the delay topics can be inside a different function. This way, it can be reused tomorrow in `editSubscription` when retries are allowed to be edited.

Reply:
agreed. done.